source/slang/slang-capabilities.capdef

```diff
@@ -383,120 +405,187 @@ alias fragmentshaderinterlock = _GL_ARB_fragment_shader_interlock | hlsl_nvapi |
 alias atomic64 = GL_EXT_shader_atomic_int64 | _sm_6_6 | cpp | cuda;
 alias atomicfloat = GL_EXT_shader_atomic_float | _sm_6_0 + hlsl_nvapi | cpp | cuda;
 alias atomicfloat2 = GL_EXT_shader_atomic_float2 | _sm_6_6 + hlsl_nvapi | cpp | cuda;
-alias groupnonuniform = GL_KHR_shader_subgroup_ballot + GL_KHR_shader_subgroup_shuffle
-                          + GL_KHR_shader_subgroup_arithmetic + GL_KHR_shader_subgroup_quad + GL_KHR_shader_subgroup_vote
-                        | _sm_6_0 | cuda;
 alias fragmentshaderbarycentric = GL_EXT_fragment_shader_barycentric | _sm_6_1;
-alias shadermemorycontrol = glsl | spirv_1_0 | _sm_5_0;
+alias shadermemorycontrol = glsl | _spirv_1_0 | _sm_5_0;
 alias shadermemorycontrol_compute = raytracingstages_compute + shadermemorycontrol;
 alias subpass = fragment + any_gfx_target;
 alias waveprefix = _sm_6_5 | _cuda_sm_7_0 | GL_KHR_shader_subgroup_arithmetic;
 alias bufferreference = GL_EXT_buffer_reference;
 alias bufferreference_int64 = bufferreference + GL_EXT_shader_explicit_arithmetic_types_int64;
 
-// Define what each HLSL shader model means on different targets.
-
-
-alias sm_4_0 = _sm_4_0
-             | glsl_spirv_1_0 + _GL_ARB_sparse_texture_clamp + _GL_EXT_samplerless_texture_functions
-             | spirv_1_0 + spvImageQuery + spvImageGatherExtended + spvMinLod + SPV_GOOGLE_user_type
+// Define what each shader model means on different targets.
+
+alias spirv_1_0 = _spirv_1_0;
+alias spirv_1_1 = _spirv_1_1
+                | spirv_1_0
+                ;
+alias spirv_1_2 = _spirv_1_2
+                | spirv_1_1
+                ;
+alias spirv_1_3 = _spirv_1_3
+                | spirv_1_2
+                ;
+alias spirv_1_4 = _spirv_1_4
+                | spirv_1_3
+                ;
+alias spirv_1_5 = _spirv_1_5 + GL_EXT_nonuniform_qualifier + GL_EXT_buffer_reference
+                | spirv_1_4
+                ;
+alias spirv_1_6 = _spirv_1_6 + GL_EXT_debug_printf + GL_EXT_demote_to_helper_invocation
+                | spirv_1_5
+                ;
+alias spirv_latest = _spirv_1_6;
+
+alias sm_4_0_version = _sm_4_0
```

Comment:
Maybe add a comment here on what `x_version` means, it means `x` for the corresponding target, and the closest match of equivvalent feature set for other targets.

Reply:
do we want to automatically translate spirv_1_0 to spirv_1_0_version when the user specifies `slangc shader.slang -profile spirv_1_0_version -target hlsl`?

Reply:
> do we want to automatically translate spirv_1_0 to spirv_1_0_version when the user specifies slangc shader.slang -profile spirv_1_0_version -target hlsl?

1. I did not add spirv_*_version but that is a good idea. 

2. We probably also should also warn if a user uses any capability defined as internal in `slangc` (`_*`)

Should '1'/'2' be another issue or part of this PR?